src/assets.rs

```diff
@@ -333,4 +347,17 @@ mod tests {
             .ok();
         assert_eq!(test.syntax_for_file("README.MD"), "Markdown");
     }
+
+    #[test]
+    fn syntax_detection_stdin_filename() {
+        let test = SyntaxDetectionTest::new();
+
+        // from file extension
+        assert_eq!(test.syntax_for_file_with_content("test.cpp", ""), "C++");
+        // from first line (fallback)
+        assert_eq!(
+            test.syntax_for_file_with_content("my_script", "#!/bin/bash"),
+            "Bourne Again Shell (bash)"
+        );
+    }
```

Comment:
This does not test the `StdIn` branch?!

Reply:
Good catch! That was pure hastiness on my part. I should have taken the changes slower to make sure I didn't lose some code paths.

I am currently working on fixing this, and I've run into an issue that I've been stuck on for a couple days. I have the following:

```rust
          fn syntax_for_stdin_with_content(&self, file_name: &str, first_line: &str) -> String {
              let input_file = InputFile::StdIn(Some(OsStr::new(file_name)));
              let syntax = self.assets.get_syntax(
                  None,
                  &input_file,
                  &mut input_file.get_reader(&io::stdin()).unwrap(),
                  &self.syntax_mapping,
              );
              syntax.name.clone()
          }
```
And both of these test cases hang:

```rust
          // from file extension
          assert_eq!(test.syntax_for_stdin_with_content("test.cpp", "a"), "C++");
          // from first line (fallback)
          assert_eq!(
              test.syntax_for_stdin_with_content("my_script", "#!/bin/bash"),
              "Bourne Again Shell (bash)"
          );
```

at the following line in fn `get_reader` :

`reader.read_until(b'\n', &mut first_line).ok();`

I discerned this from println statement debugging.
And I can't figure out why.
I didn't change this line "`&mut input_file.get_reader(&io::stdin()).unwrap()`" , nor can I see anything that changed in the function get_reader or even the input_file.rs itself.

I saw you pushed some commits too, but I looked at all of those and am not seeing anything which changed. Any ideas? Thanks for your patience with my newbness!

Reply:
Also worth noting that manual tests seem to work fine for the syntax detection from filename extension and from first line using stdin. So it must be something specific to how the test case is setup.

Reply:
Well, these cases don't hang. They are waiting for input from STDIN :smile:. If you type something on the terminal and press enter (twice), the tests will keep on going and fail or succeed depending on what you input. 

That's obviously not how we want to test this in a unit/integration test.

I have pushed a commit that changes the way that `stdin` is passed into the `get_reader` function. This way, we can just pass in a byte-string as content (instead of reading from *actual* stdin).

Reply:
ahhh, I see now!! Thanks!